src/libraries/System.Private.CoreLib/src/System/Reflection/NullabilityInfoContext.cs

```diff
@@ -204,12 +206,7 @@ public NullabilityInfo Create(PropertyInfo propertyInfo)
 
             if (setter != null)
             {
-                if (IsPrivateOrInternalMethodAndAnnotationDisabled(setter))
-                {
-                    nullability.WriteState = NullabilityState.Unknown;
-                }
-
-                CheckNullabilityAttributes(nullability, setter.GetParameters()[0].GetCustomAttributesData());
+                CheckNullabilityAttributes(nullability, setter.GetParameters()[^1].GetCustomAttributesData());
```

Comment:
Change here is because indexer properties have multiple parameters. The last parameter is the one we're interested in (`value`).